lib/ansible/modules/system/ufw.py

```diff
@@ -327,6 +349,8 @@ def main():
         supports_check_mode=True,
         mutually_exclusive=[
             ['name', 'proto', 'logging'],
+            ['direction', 'interface_in'],
+            ['direction', 'interface_out'],
```

Comment:
You need two more entries `['interface', 'interface_in']` and `['interface', 'interface_out']`.

Reply:
Is that required even though `interface` depends on `direction`?

https://github.com/ansible/ansible/blob/0d34a2eceea9178553e23f53ed24ba58e3722833/lib/ansible/modules/system/ufw.py#L332-L334

I couldn't find much documentation on `required_by` other than a one-sentence description in `developing_modules_general_windows.rst`, but my interpretation is that an error will always be raised if the param in the key is given without the param(s) in the value?  If not I'll fix it.

Thanks for the review!

Reply:
You are right, they are not required because of that. Also, `required_by` works as you describe (I checked its implementation to be sure: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/validation.py#L133-L166). But maybe adding a comment that `required_if` implies that `interface` is also mutually exclusive to both `interface_in` and `interface_out` would probably help future readers as well :)

Reply:
Done!  I also added a note about exclusivity to the module documentation for `interface` (it was already there for `direction`).